### PR TITLE
Add list view for events

### PR DESCRIPTION
**Pull request by jszwedko**
This ended up being way more changes than I intended, but hopefully it is still
reviewable! Please ask questions if anything is unclear.

Big changes:
* Create EventList view component with EventListItem subcomponents. This
  will display whatever events were on the map as a list by using the
  bounds of the map to query for the events. This could probably
  refactored to not re-query for events, but pull them up into the
  Events component state. Mostly unstyled for now.
* Update the map state in the query part of the URL. This allows the user to
	come back to the same part of the map after clicking on an event and is needed
	to make sure the added list view shows the right events.
* Created Events component to wrap the map and list components and add
  an additional bar to switch between the both. Currently mostly
  unstyled. Nest the routes for the event list and map under this as
  they will only appear in this view.
* Created simple store to store the user's lat/lng for use computing the
  distance to events for the list view. In the future, we expect to have
  a search bar to search by address. This will be a sibling component to
  the map and list so simplest way to share this state seems to be
  pulling it out in this fashion. I don't feel we are storing enough
  state to need to involve Vuex yet.

Medium changes:
* Flatten publisher event results. Was previously returning an array of arrays
  of events. We only have one publisher right now so this is OK, but would have
  caused a suprising bug as soon as a second was added.

Small changes:
* Refactor styling of "full screen" containers to .container since every
  parent element of the map needs to have 100% height and no margins
* Pull out phone number formatting filter since it is now used in two
  places (on the individual event view and the list view)
* Pull up .section style as it is now used on list view and individual
  event view
* Guard against nulls in phone number format filter
* Rename NoticesMap to EventsMap to match the convention of the other
  component names
* Parse zoom as an integer as the leaflet map component expects it to be so

Some remaining TODOs, but wanted to get this much reviewed before I went too
much further.

### CODE REVIEW

**Comment by jszwedko** [comment]
Bump @aekong @akegan 

I realize this is a lot of changes. I'm happy to walk through if it helps.

File: src/components/Events.vue [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<template>
```
Comment (akegan) [inline src/components/Events.vue:1]: I'm a bit confused about what "Events" is vs EventList, and EventListItem, and EventMap. Is this an individual event page?

Comment (akegan) [reply src/components/Events.vue:1]: ah, it's the container for an event. does the endpoint events need to exist without passing in an event ID?

Comment (jszwedko) [reply src/components/Events.vue:1]: 👍 this is basically a container to hold either the map or the list, along with other components that show in both views (e.g. the toggle buttons). I could probably pick a better name.

**Review by akegan — APPROVED** [review]

**Pull request merged**